nil/node.py

```diff
@@ -0,0 +1,83 @@
+import sys
+from collections import defaultdict
+
+import click
+from addict import Dict
+from aiohttp import web
+from loguru import logger
+from mode import Service, Worker
+
+from .injector import inject, injector
+from .interface import Message, UserMessage
+from .logging import configure_logging
+from .network.interface import AbstractNetwork
+from .storage.data.interface import AbstractDataStorage
+from .storage.hash.interface import AbstractHashStorage
+from .web_server import WebServer
+
+
+class Node(Service):
+
+    @inject
+    def __init__(self, data_storage: AbstractDataStorage, hash_storage: AbstractHashStorage,
+                 network: AbstractNetwork, web_server: WebServer):
+        super().__init__()
+        self.data_storage = data_storage
+        self.hash_storage = hash_storage
+        self.network = network
+        self.web_server = web_server
+
+    def on_init_dependencies(self):
+        return [
+            self.data_storage,
+            self.hash_storage,
+            self.network,
+            self.web_server,
+        ]
+
+    async def on_start(self):
+        router = self.web_server.app.router
+        router.add_route("POST", "/send_message", self.send_message)
+        router.add_route("POST", "/get_incoming_messages", self.get_incoming_messages)
+
+    async def send_message(self, request):
+        data = await request.json()
+        message = UserMessage.from_dict(data)
+        message = Message(
+            data=message.data,
+            source=self.network.identity,
+            destination=message.address,
+        )
+        message.tx = await self.hash_storage.put_payload(message.checksum)
+        await self.network.send_message(message)
+        await self.data_storage.put_message(message)
+        return web.json_response()
+
+    async def get_incoming_messages(self, request):
+        messages = await self.network.get_incoming_messages()
+        for message in messages:
+            await self.data_storage.put_message(message)
+        messages = [UserMessage(data=m.data, address=m.source) for m in messages]
+        return web.json_response([m.to_dict() for m in messages])
+
+
+@click.command(context_settings=dict(auto_envvar_prefix="NIL"))
+@click.option("--log-level", show_default=True, default="INFO")
+@click.option("--web-host", show_default=True, default="localhost")
+@click.option("--web-port", show_default=True, default=8080, type=int)
+@click.option("--io-network-max-workers", show_default=True, default=100, type=int)
+@click.option("--io-file-max-workers", show_default=True, default=5, type=int)
+# data storage
+@click.option("--data-storage-type", show_default=True, default="sqlite")
```

Comment:
Maybe add option validation, given, that this value is used in `importlib.import_module` later on?

```suggestion
@click.option("--data-storage-type", show_default=True, default="sqlite", type=click.Choice(["sqlite"]))
```